terraform/environment/opensearch.tf

```diff
@@ -0,0 +1,114 @@
+data "aws_default_tags" "current" {
+  provider = aws.eu_west_1
+}
+
+data "aws_caller_identity" "current" {
+  provider = aws.eu_west_1
+}
+
+data "aws_vpc" "main" {
+  filter {
+    name   = "tag:application"
+    values = [data.aws_default_tags.current.tags.application]
+  }
+  provider = aws.eu_west_1
+}
+
+data "aws_availability_zones" "available" {
+  state    = "available"
+  provider = aws.eu_west_1
+}
+
+data "aws_subnet" "application" {
+  count             = 3
+  vpc_id            = data.aws_vpc.main.id
+  availability_zone = data.aws_availability_zones.available.names[count.index]
+
+  filter {
+    name   = "tag:Name"
+    values = ["application*"]
+  }
+  provider = aws.eu_west_1
+}
+
+resource "aws_opensearchserverless_security_policy" "lpas_collection_encryption_policy" {
+  name        = "policy-${local.environment_name}"
```

Comment:
these resources are missing providers and will fail to create on the pipeline (as there's no default provider, the user in identity will try to create these in us-east-1 but it doesn't have permission to do that)